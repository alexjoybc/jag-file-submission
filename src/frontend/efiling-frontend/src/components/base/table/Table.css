table {
  width: 100%;
}

th {
  padding: 3px;
}

td {
  vertical-align: top;
  padding: 3px;
  white-space: pre-wrap;
}
