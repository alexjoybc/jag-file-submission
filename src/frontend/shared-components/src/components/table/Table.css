table {
  width: 100%;
}

td {
  vertical-align: top;
  padding: 3px;
  white-space: pre-wrap;
}

.blue-stripe tbody tr:nth-child(even) {
  background: rgba(241, 244, 250, 1);
}

.blue-stripe tbody tr:nth-child(odd) {
  background: rgba(225, 232, 242, 1);
}
