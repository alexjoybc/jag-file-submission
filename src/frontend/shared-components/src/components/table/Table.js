import React from "react";
import PropTypes from "prop-types";

import "./Table.css";

const TableElement = ({ element: { name, value } }) => {
  return (
    <tr>
      <td>{name}</td>
      <td>{value}</td>
    </tr>
  );
};

export const Table = ({ heading, elements, styling }) => {
  const tableComponents = elements.map(element => {
    return <TableElement key={element.key || element.name} element={element} />;
  });

  return (
    <table className={styling}>
      <thead>
        <tr>
          <th colSpan="2">{heading}</th>
        </tr>
      </thead>
      <tbody>{tableComponents}</tbody>
    </table>
  );
};

TableElement.propTypes = {
  element: PropTypes.shape({
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
  }).isRequired
};

Table.propTypes = {
  heading: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  elements: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
        .isRequired
    }).isRequired
  ),
  styling: PropTypes.string
};

Table.defaultProps = {
  styling: "",
  heading: "",
  elements: []
};
