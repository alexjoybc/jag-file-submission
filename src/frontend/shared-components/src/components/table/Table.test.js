import React from "react";
import { render } from "@testing-library/react";

import Table from "./Table";

describe("Table Component", () => {
  test("Matches the snapshot", () => {
    const tableElements = [
      { name: "BCeID", value: "bobross42" },
      {
        name: "Last Name",
        value: "Ross"
      },
      { name: "First Name", value: "Bob" },
      { name: "Email Address", value: "bob.ross@pbs.com" }
    ];

    const table = {
      header: "BCeID Info",
      tableElements
    };

    const { asFragment } = render(<Table table={table} />);
    expect(asFragment).toMatchSnapshot();
  });
});
