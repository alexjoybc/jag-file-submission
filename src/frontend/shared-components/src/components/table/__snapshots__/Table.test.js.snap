// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Matches the blue striped snapshot 1`] = `[Function]`;

exports[`Table Component Matches the default snapshot 1`] = `[Function]`;
